packages/shared/lib/utils/utils.ts

```diff
@@ -239,3 +239,7 @@ export function getConnectionConfig(queryParams: any): Record<string, string> {
     const arr = Object.entries(queryParams).filter(([, v]) => typeof v === 'string'); // Filter strings
     return Object.fromEntries(arr) as Record<string, string>;
 }
+
+export function encodeParameters(params: Record<string, any>): Record<string, string> {
+    return Object.fromEntries(Object.entries(params).map(([key, value]) => [key, encodeURIComponent(String(value))]));
```

Comment:
isn't this === to `new URLSearchParams(params).toString()`

Reply:
@bodinsamuel  `new URLSearchParams(params).toString()` and `Object.fromEntries(Object.entries(params).map(([key, value]) => [key, encodeURIComponent(String(value))]))` are similar but handle encoding differently.

- `URLSearchParams(params).toString()`: Encodes parameters for query strings but may not handle all special characters as thoroughly.
- `encodeURIComponent`: Provides more comprehensive encoding of special characters.

Reply:
Yes they are the same, but in our code, `encodeParameters` is used to prepare the `connectionConfig` object for interpolation into a URL. This is necessary because `interpolateStringFromObject` expects an object with pre-encoded values, ensuring that when these values are interpolated into a URL, they do not introduce any invalid characters or break the URL format.